File: 30_days_Python_HackerRank_Challenge/LinkedList.py
#python-2
class Node:
    def __init__(self,data):
        self.data = data
        self.next = None
class Solution:
    def display(self,head):
        current = head
        while current:
            print current.data,
            current = current.next
            
    def insert(self, head, data):
        """
        Inserts a new node with 'data' at the tail of the list.
        """
        new_node = Node(data)
        
        # Scenario 1: The list is empty
        if head is None:
            return new_node
        
        # Scenario 2: The list has elements
        current = head
        # Walk to the last node (the one where .next is None)
        while current.next:
            current = current.next
        
        # Link the new node to the end
        current.next = new_node
        
        return head
            
            
mylist= Solution()
T=int(input())
head=None
for i in range(T):
    data=int(input())
    head=mylist.insert(head,data)    
mylist.display(head);

File: 30_days_C(modern)_HackerRank_Challenge/LinkedList.c
#include <stdlib.h>
#include <stdio.h>	
typedef struct Node
{
    int data;
    struct Node* next;
}Node;


struct Node* insert(struct Node* head, int data) {
    struct Node* newNode = (struct Node*)malloc(sizeof(struct Node));
    newNode->data = data;
    newNode->next = NULL;

    if (head == NULL)
        return newNode;

    struct Node* current = head;
    while (current->next != NULL)
        current = current->next;

    current->next = newNode;

    return head;
}

void display(Node *head)
{
	Node *start=head;
	while(start)
	{
		printf("%d ",start->data);
		start=start->next;
	}
}
int main()
{
	int T,data;
    scanf("%d",&T);
    Node *head=NULL;	
    while(T-->0){
        scanf("%d",&data);
        head=insert(head,data);
    }
  display(head);
		
}
